### ISSUE TITLE
Ability to specify revision for win_gitrepos undocumented

### ISSUE THREAD

**Issue by tkent-xetus** [saltstack/salt#24268]
We recently ended up in a situation where our win_repo couldn't be updated due to a recent syntax error introduced into the official salt-winrepo. We needed to either use a different repository or use a previous revision of the salt-winrepo repository to avoid the problem.

After looking through the code, it turns out that salt supports specifying a specific revision to use for a `win_gitrepos` entry, probably added to defend against the situation we ran into. All we had to do was find the revision before the problem was introduced and update our config like so:

```
win_gitrepos:
  - '<commit_id> https://github.com/saltstack/salt-winrepo.git'
```

It be great if this feature was documented

**Comment by rallytime** [comment]
Thanks for the report @tkent-xetus. We should definitely get this feature documented. Can you also post the output of `salt <windows-minion> test.versions_report`? That way we know what "recent syntax error introduced into official salt-winrepo" refers to.

@twangboy

**Comment by tkent-xetus** [comment]
@rallytime 

Absolutely and I'll do ya one better. Below is the output from our versions report, but we also found/fixed the salt-winrepo issue and [submitted a pull request](https://github.com/saltstack/salt-winrepo/pull/214). It turns out the issue was just some YAML syntax mistake in a winrepo entry, nothing specific to our salt setup.

As always - really appreciate the quick responses!

```
gd-master.qa.xetusone.com:
               Salt: 2015.5.0
             Python: 2.7.8 (default, Jun 30 2014, 16:08:48) [MSC v.1500 64 bit (AMD64)]
             Jinja2: 2.7.3
           M2Crypto: 0.21.1
     msgpack-python: 0.4.5
       msgpack-pure: Not Installed
           pycrypto: 2.6
            libnacl: Not Installed
             PyYAML: 3.11
              ioflo: Not Installed
              PyZMQ: 14.5.0
               RAET: Not Installed
                ZMQ: 4.0.5
               Mako: Not Installed
saltmaster.internal.xetus.com:
                      Salt: 2015.5.0
                    Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
                    Jinja2: 2.7.2
                  M2Crypto: 0.21.1
            msgpack-python: 0.3.0
              msgpack-pure: Not Installed
                  pycrypto: 2.6.1
                   libnacl: Not Installed
                    PyYAML: 3.10
                     ioflo: Not Installed
                     PyZMQ: 14.0.1
                      RAET: Not Installed
                       ZMQ: 4.0.4
                      Mako: 0.9.1
     Debian source package: 2015.5.0+ds-1trusty1
```

**Comment by rallytime** [comment]
Awesome! Thanks for submitting a pull request! Once that is merged in, would you consider this issue okay to close?

**Comment by tkent-xetus** [comment]
Ack, sorry for not being clear. I'll be a bit more long winded this time.

We didn't file the salt-winrepo issue because we assumed our pull request would get picked up and square that issue away. After all, it was just a simple syntax mistake.

This issue is just about the documentation for `win_gitrepos`. As documented, the `win_gitrepos` configuration doesn't support specifying a branch or tag or commit number. Therefore, you are kinda crossing your fingers that the maintainers of the repositories you use always keeps them in a working state. This does make a bit of sense due to the fact that this is being treated as a "live" mapping of software packages (like an APT repo), but it's risky too. As with the recent salt-winrepo issue, things can and do go wrong.

Looking at the documentation, if a repo you're using introduces a bug, your only options to get around the problem is to either:

1. Wait until the maintainers fix it.
2. Create/host a new repository from the last working commit of the broken repository, and update your configurations to use that.

These options aren't terrible, but they're not that great. However, what the documentation doesn't say is that a simple/effective option is already in place - the (undocumented) syntax I described above. Since you have the ability to specify a commit number with `win_gitrepos`, you just have to specify the last working commit until the maintainer fixes the issue, then remove it once they have.


So basically, you've already got a pretty much perfect solution available when maintainers break `win_gitrepos`, it's just that people who don't go through the source won't know about it.

**Comment by rallytime** [comment]
Oh, I understand now. Thanks for the clarification!

**Comment by twangboy** [comment]
Jacob added the documentation.
https://github.com/saltstack/salt/pull/24468/files#diff-e9eedf3de8950153e9c807004a007782R2436

**Comment by tkent-xetus** [comment]
@twangboy Thanks!

**Comment by twangboy** [comment]
@tkent-xetus Thanks for pointing out the glaring lack of documentation.
